File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.Result;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RequestMapping("/brand")
@RestController//这个注解是Controller和ResponseBody的结合
public class BrandController {

    @Reference
    private BrandService brandService;
    //查询所有
    @RequestMapping("findAll")
    public List<TbBrand> findAll(){

        return brandService.findAll();
    }

   //条件查询+分页实现
    @RequestMapping("/search")
    public PageResult findPageByNameAndChar(@RequestBody TbBrand brand,Integer pageNum, Integer pageSize){
        return brandService.findPageByNameAndChar(brand,pageNum,pageSize);
    }
    /**
     * 添加
     */
    @RequestMapping("/add")
    public Result addBrand(@RequestBody TbBrand brand){
        try {
            brandService.addBrand(brand);
            return new Result(true,"添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"添加成功");
        }
    }
    /**
     * 通过id查找信息
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }
    /**
     * 修改brand
     */
    @RequestMapping("/update")
    public Result updateBrand(@RequestBody TbBrand brand){
        try {
            brandService.updateBrand(brand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改成功");
        }
    }
    /**
     * 批量删除
     */
    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除成功");
        }
    }
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
app.controller("brandController",function ($scope,$controller,brandService) {

    //控制器继承代码  参数一：继承的父控制器名称  参数二：固定写法，共享$scope对象
    $controller("baseController",{$scope:$scope});

    //定义查询所有品牌列表的方法
    $scope.findAll=function () {
        //response接受相应结果
        brandService.findAll().success(function (response) {
            $scope.list = response;
        })
    }

    //条件查询
    $scope.searchEntity={};//解决初始化参数为空的情况
    $scope.search=function (pageNum,pageSize) {
        brandService.search($scope.searchEntity,pageNum,pageSize).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems=response.total;
        })
    }
    //添加品牌
    $scope.save=function () {
        var method=null;
        if($scope.entity.id!=null){
            //修改
            method=brandService.update($scope.entity)
        }else{
            //添加
            method=brandService.add($scope.entity)
        }
        method.success(function (response) {
            if(response.success){
                //添加成功
                $scope.reloadList();
            }else{
                alert(response.message);
            }
        })
    }
    //根据id查询品牌数据
    $scope.findOne=function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity=response;
        })
    }

    //删除操作$http.get("../brand/delete.do?ids="+$scope.selectIds)
    $scope.delete=function () {
        if(confirm("您确定要删除吗")){
            brandService.delete($scope.selectIds).success(function (response) {
                if (response.success){
                    //删除成功,从新加载列表
                    $scope.reloadList();
                }else{
                    alert(response.message);
                }
            })
        }
    }


})